packages/relay/src/lib/eth.ts

```diff
@@ -1750,13 +1750,21 @@ export class EthImpl implements Eth {
       EthImpl.ethGetTransactionReceipt,
       requestIdPrefix,
     );
+
     if (cachedLog) {
+      const block = await this.getBlockByHash(cachedLog.blockHash, false);
+      const gasPriceForTimestamp = await this.getFeeWeibars(
+        EthImpl.ethGetTransactionReceipt,
+        requestIdPrefix,
+        block?.timestamp,
+      );
+
       const receipt: any = {
         blockHash: cachedLog.blockHash,
         blockNumber: cachedLog.blockNumber,
         contractAddress: cachedLog.address,
         cumulativeGasUsed: EthImpl.zeroHex,
-        effectiveGasPrice: EthImpl.zeroHex,
+        effectiveGasPrice: numberTo0x(gasPriceForTimestamp),
```

Comment:
If it's not too much scope creep please consider fixing this for the setting of `effectiveGasPrice` further down should it be null after https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/eth.ts#L1794

Then similar ask, add an acceptance tests that checks a submitted ContractCall that executes a transaction and you confirm the `effectiveGasPrice` is correct.
This one should be easier as you can use the details from [mirrorContractDetails.hash](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/server/tests/acceptance/rpc_batch1.spec.ts#L83) to make the query